stock_barcodes/wizard/stock_barcodes_read.py

```diff
@@ -0,0 +1,745 @@
+        for option in options_required:
+            if not getattr(self, option.field_name, False):
+                if option.field_name == "lot_id" and self.product_id.tracking == "none":
+                    continue
+                self.step = option.step
+                break
+        if not self.step:
+            self.step = options_required[:1].step
+
+        options = self.option_group_id.option_ids.filtered(
+            lambda op: op.step == self.step and op.to_scan
+        )
+        self._set_messagge_info(
+            "info", _("Scan {}").format(", ".join(options.mapped("name")))
+        )
+
+    @api.onchange("package_id")
+    def onchange_package_id(self):
+        if self.manual_entry:
+            self.barcode = self.package_id.name
+            self.process_barcode_package_id()
+
+    def action_confirm(self):
+        if not self.check_option_required():
+            return False
+        record = self.browse(self.ids)
+        record.write(self._convert_to_write(self._cache))
+        self = record
+        res = self.action_done()
+        self.invalidate_recordset()
+        self.play_sounds(res)
+        self._set_focus_on_qty_input()
+        return res
+
+    def process_lot_before_done(self):
+        if (
+            not self.lot_id
+            and self.lot_name
+            and self.product_id
+            and self.product_id.tracking != "none"
+            and self.option_group_id.create_lot
+        ):
+            self.lot_id = self._create_new_lot()
+        return True
+
+    def play_sounds(self, res):
+        if res:
+            self.env["bus.bus"]._sendone(
+                "barcode_scan",
+                "stock_barcodes_sound",
+                {"sound": "ok"},
+            )
+        else:
+            self.env["bus.bus"]._sendone(
+                "barcode_scan",
+                "stock_barcodes_sound",
+                {"sound": "ko"},
+            )
+
+    def _set_focus_on_qty_input(self, field_name=None):
+        if field_name is None:
+            field_name = "product_qty"
+        if field_name == "product_qty" and self.packaging_id:
+            field_name = "packaging_qty"
+        self.env["bus.bus"]._sendone(
+            "barcode_scan",
+            "stock_barcodes_focus",
+            {"field_name": field_name},
+        )
+
+    @api.onchange("product_id")
+    def onchange_product_id(self):
+        self.product_uom_id = self.product_id.uom_id
+
+    @api.onchange("manual_entry")
+    def onchange_manual_entry(self):
+        if self.manual_entry and self.option_group_id.manual_entry_field_focus:
+            self._set_focus_on_qty_input(self.option_group_id.manual_entry_field_focus)
+
+    def _prepare_lot_vals(self):
+        return {
+            "name": self.lot_name,
+            "product_id": self.product_id.id,
+            "company_id": self.env.company.id,
+        }
+
+    def _create_new_lot(self):
+        StockProductionLot = self.env["stock.production.lot"]
```

Comment:
Hi, you need to change this model to stock.lot

        StockProductionLot = self.env["stock.lot"]